File: CodeJava/Solve.java
package CodeJava;

public class Solve {
    public String customSortString(String order, String s) {
        String res = "";
        for (int i = 0; i < order.length(); i++) {
            String str = order.charAt(i)+ "";
            while (isIn(s , str)) {
                s = s.replace(str, "");
                res = res + str;
            }
        }
        return res + s;
    }

    public boolean isIn(String s,String str){
        boolean is = false;

        for (int i = 0; i < s.length(); i++) {
            String p = s.charAt(i) + "";
            if ( p.equals(str)) {
                is = true;
            }
        }
        return is;
    }
}

File: CodeJavascript/Solve.js

//leetcode solution
//Problem: https://leetcode.com/problems/custom-sort-string/

function customSortString(order, s) {
    let res = "";
    for (let i = 0; i < order.length; i++) {
        let str = order.charAt(i);
        if (isIn(s, str)) {
            let m = howMany(s,str);
            while (m>0) {
                res = res +
                m--;     
            }
            s = s.replaceAll(str,"");
            res = res + str;
        }
    }
    return res + s;
}

function isIn(s, str) {
    let is = false;
    for (let i = 0; i < s.length; i++) {
        let p = s.charAt(i);
        if (p == str) {
            is = true;
        }
    }
    return is;
}
function howMany(s,str) {
    let is = 0;
    for (let i = 0; i < s.length; i++) {
        let p = s.charAt(i);
        if (p == str) {
            is++;
        }
    }
    return is;
}
// console.log(customSortString("cba","abcd"));
export {customSortString};